src/website/utils/fields.py

```
@@ -2,22 +2,68 @@
+                # For PNG, convert to RGB to allow optimization
+                if img.format.upper() == "PNG" and img.mode in ("RGBA", "P"):
+                    img = img.convert("RGB")
+
+                # Save the compressed image to the buffer
+                img.save(buffer, format=img.format, **save_kwargs)
+
+                # Replace file content with compressed content
+                file.file = ContentFile(buffer.getvalue())
+
+    except Exception as e:
+        raise ValidationError(f"Invalid image file: {file.name}. Error: {e}")
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)